components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/NetworkMessageEncoder.cs

```diff
@@ -74,7 +74,8 @@ public class NetworkMessageEncoder : IMessageEncoder {
                 return Task.FromResult(result);
```

Comment:
should we add result.ToList() here to ensure that the enumeration was iterated and the exception (if any) occurs directly and could be catched?

Reply:
I will have to use another transformation then to return the correct type, `result.ToList().AsEnumerable()`. So I would say that we keep it as it is to avoid those transformation on each encoding calls. What do you think ?

Reply:
A list is also enumerable...

Reply:
OK. I had a problem with wrapping it in the `Task.FromResult()`. It is done now.